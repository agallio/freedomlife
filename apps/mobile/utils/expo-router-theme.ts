import { DefaultTheme, DarkTheme, type Theme } from '@react-navigation/native'

const fonts = {
  regular: {
    fontFamily: 'inter',
  },
  medium: {
    fontFamily: 'inter-medium',
  },
  bold: {
    fontFamily: 'inter-bold',
  },
  heavy: {
    fontFamily: 'inter-bold',
  },
} as Theme['fonts']

export const CustomLightTheme = {
  ...DefaultTheme,
  colors: {
    ...DefaultTheme.colors,
    primary: '#6ee7b7',
    background: '#f3f4f6',
    card: '#f3f4f6',
  },
  fonts,
}

export const CustomDarkTheme = {
  ...DarkTheme,
  colors: {
    ...DarkTheme.colors,
    primary: '#047857',
    background: '#1f2937',
    card: '#1f2937',
  },
  fonts,
}
