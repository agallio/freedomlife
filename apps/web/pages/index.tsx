// Screen
import HomeScreen from '@repo/app/features/home'

// Icon Components
import FreedomLifeIcon from '@repo/app/components/icons/freedomlife-icon'

export default function HomePage() {
  return (
    <div className="mx-auto flex w-full max-w-sm flex-col px-6 pb-28 pt-4 sm:max-w-md">
      <div className="flex pb-4">
        <FreedomLifeIcon className="w-[230px]" />
      </div>

      <HomeScreen />
    </div>
  )
}
