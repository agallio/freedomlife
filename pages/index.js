import React, { Component } from 'react';
import { connect } from 'react-redux';
import Router from 'next/router';
import {
  LinearProgress,
  Card,
  CardContent,
  Typography,
  Grid,
  Fab
} from '@material-ui/core';
import moment from 'moment';
import 'moment/locale/id';

import { fetchGuideToday } from '../src/actions/guide';
import { fetchTodayChapter } from '../src/actions/bible';

class Index extends Component {
  componentDidMount = async () => {
    if (this.props.guide.guideToday.date !== moment().format('DD-MM-YYYY')) {
      this.props.fetchGuideToday();
    }
  };

  toBible = () => {
    if (
      this.props.guide.guideToday.date === moment().format('DD-MM-YYYY') &&
      this.props.bible.chapters.passage.length === 0
    ) {
      this.props.fetchTodayChapter();
    }
    Router.push('/bible');
  };

  render() {
    const { isFetching, guideToday } = this.props.guide;

    return (
      <div>
        {isFetching && <LinearProgress color="secondary" />}
        <div className="container" style={{ paddingBottom: 90 }}>
          <h2 className="header-title" style={{ marginTop: 70 }}>
            Freedom Life
          </h2>
          <p className="header-subtitle">
            Aplikasi panduan baca Alkitab setahun
          </p>

          <div style={{ marginTop: 40 }}>
            <Card className="styled-card">
              <CardContent>
                <Typography className="bold-text primary" variant="h5">
                  Panduan Hari Ini
                </Typography>
                <Typography className="light-text primary" variant="subtitle1">
                  {moment().format('dddd, LL')}
                </Typography>

                {['PL', 'PB1', 'PB2'].map(item => (
                  <Grid
                    key={item}
                    container
                    direction="row"
                    alignItems="center"
                    spacing={2}
                    style={{ marginTop: 5 }}
                  >
                    <Grid item xs={3} sm={2} md={2}>
                      <div className="guide-passage-box">
                        <h5 className="guide-passage-box-text">{item}</h5>
                      </div>
                    </Grid>
                    <Grid item xs={9} sm={10} md={10}>
                      {isFetching ? (
                        <LinearProgress />
                      ) : (
                        <Typography
                          className="bold-text primary"
                          variant="h6"
                          style={{ fontSize: 17 }}
                        >
                          {item === 'PL'
                            ? guideToday.pl_name
                            : item === 'PB1'
                            ? guideToday.pb1_name
                            : item === 'PB2'
                            ? guideToday.pb2_name
                            : 'Tidak ada data'}
                        </Typography>
                      )}
                      <Typography
                        className="light-text primary"
                        variant="subtitle1"
                      >
                        {item === 'PL'
                          ? 'Perjanjian Lama'
                          : item === 'PB1'
                          ? 'Perjanjian Baru 1'
                          : item === 'PB2'
                          ? 'Perjanjian Baru 2'
                          : ''}
                      </Typography>
                    </Grid>
                  </Grid>
                ))}

                <br />
                <Fab
                  className="guide-passage-box-fab"
                  size="small"
                  variant="extended"
                  color="primary"
                  onClick={this.toBible}
                >
                  Baca
                </Fab>
              </CardContent>
            </Card>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    guide: state.guide,
    bible: state.bible
  };
};

const mapDispatchToProps = dispatch => {
  return {
    fetchGuideToday: () => dispatch(fetchGuideToday()),
    fetchTodayChapter: version => dispatch(fetchTodayChapter(version))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Index);
