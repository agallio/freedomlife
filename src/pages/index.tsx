import { useEffect, useState } from 'react'
import Router from 'next/router'
import { AnimatePresence } from 'framer-motion'

import JumboHeader from '@/components/JumboHeader'
import HomeBox from '@/components/Home/HomeBox'
import NewYearDialog from '@/components/Home/NewYearDialog'
import NewUserBox from '@/components/Home/NewUserBox'

import { useDispatchGuide } from '../store'

import dayjs from '@/utils/dayjs'
import useRequest from '@/utils/hooks/useRequest'

import type { ApiResponse, GuideDataResponse } from '@/types/api'

const Home = (): JSX.Element => {
  const guideDispatch = useDispatchGuide()
  const [open, setOpen] = useState(false)

  const { data } = useRequest<ApiResponse<GuideDataResponse>>({
    url: `/api/guide/${dayjs().format('DD-MM-YYYY')}`,
  })

  const toBible = () => {
    guideDispatch({ type: 'SET_GUIDE_DATE', data: '' })
    guideDispatch({ type: 'SET_GUIDE_PASSAGE', data: '' })
    Router.push('/read?guide=true')
    document.body.scrollTop = 0
    document.documentElement.scrollTop = 0
  }

  useEffect(() => {
    if (open) {
      document.body.style.overflow = 'hidden'
    } else {
      document.body.style.overflow = 'unset'
    }
  }, [open])

  return (
    <div className="max-w-md p-6 mx-auto mb-20 sm:py-6 md:mb-16 landscape:mx-auto">
      <JumboHeader isHome />

      <main>
        <HomeBox data={data} toBible={toBible} />

        <NewUserBox />

        <AnimatePresence>
          {open && <NewYearDialog handleClose={() => setOpen(false)} />}
        </AnimatePresence>
      </main>
    </div>
  )
}

export default Home
