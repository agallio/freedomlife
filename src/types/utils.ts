import type { AxiosError } from 'axios'
import type { UseQueryResult } from 'react-query'

export interface BibleList {
  name: string
  abbr: string
  passage: number
}

// Fetched Hooks
export interface QueryResult<T> {
  data: T
  error: AxiosError<never> | null
  isError: boolean
  isLoading: boolean
  refetch: (options?: {
    throwOnError: boolean
    cancelRefetch: boolean
  }) => Promise<UseQueryResult>
}
