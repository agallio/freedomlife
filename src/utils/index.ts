import * as gtag from './gtag'
import dayjs from './dayjs'
import useRequest from './hooks/useRequest'
import useFetchedGuide from './hooks/useFetchedGuide'

export { gtag, dayjs, useRequest, useFetchedGuide }
