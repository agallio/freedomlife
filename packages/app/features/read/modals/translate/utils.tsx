import { bibleTranslations } from '../../../../utils/constants'

export function filterTranslationsWithLookupSet(
  availableBibleTranslations?: string[],
) {
  const lookupSet = new Set(availableBibleTranslations)

  return bibleTranslations
    .map((item) => ({
      ...item,
      versions: item.versions.filter((version) => lookupSet.has(version.key)),
    }))
    .filter((item) => item.versions.length > 0)
}
