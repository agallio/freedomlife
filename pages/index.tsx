import Router from 'next/router'
import {
  Card,
  CardContent,
  Fab,
  Fade,
  Grid,
  LinearProgress,
  Typography,
} from '@material-ui/core'
import { Skeleton } from '@material-ui/lab'

import { dayjs, useRequest } from '../src/utils'
import { useDispatchGuide } from '../src/store'

// Types
import type { ApiResponse, GuideDataResponse } from '../src/types'

export const Home: React.FC = () => {
  const guideDispatch = useDispatchGuide()

  const { data } = useRequest<ApiResponse<GuideDataResponse>>({
    url: '/api/guide/today',
  })

  const toBible = () => {
    guideDispatch({ type: 'SET_GUIDE_DATE', data: '' })
    Router.push('/bible')
  }

  return (
    <>
      {!data && <LinearProgress color="secondary" />}
      <Fade in>
        <div className="container" style={{ paddingBottom: 90 }}>
          <h2
            className="header__title"
            data-testid="headerTitle"
            style={{ marginTop: 65 }}
          >
            FreedomLife
          </h2>
          <p className="header__subtitle" data-testid="headerSubtitle">
            Aplikasi Panduan Baca Alkitab Setahun
          </p>

          <div style={{ marginTop: 40 }}>
            <Card className="homecard">
              <CardContent>
                <Typography
                  className="homecard__title"
                  data-testid="homecardTitle"
                  style={{ fontSize: '1.5rem !important', lineHeight: 1 }}
                >
                  Panduan Hari Ini
                </Typography>
                <Typography
                  className="homecard__subtitle"
                  data-testid="homecardSubtitle"
                  variant="subtitle1"
                >
                  {dayjs().format('dddd, DD MMMM YYYY')}
                </Typography>

                {['PL', 'PB', 'ALT'].map((item) => (
                  <Grid
                    key={item}
                    container
                    direction="row"
                    alignItems="center"
                    spacing={2}
                    style={{ marginTop: 5 }}
                  >
                    <Grid item xs={3} sm={2} md={2}>
                      <div className="homecard__pbox">
                        <p
                          className="homecard__pbox__text"
                          data-testid={`homecardPBox_${item}`}
                        >
                          {item}
                        </p>
                      </div>
                    </Grid>
                    <Grid item xs={9} sm={10} md={10}>
                      {!data ? (
                        <Skeleton variant="text" animation="wave" height={25} />
                      ) : (
                        <Typography
                          className="homecard__title"
                          variant="h6"
                          style={{ fontSize: 17 }}
                        >
                          {item === 'PL'
                            ? data.data?.pl_name
                            : item === 'PB'
                            ? data.data?.pb_name
                            : item === 'ALT'
                            ? data.data?.alt_name
                            : 'Tidak ada data'}
                        </Typography>
                      )}
                      <Typography
                        className="homecard__subtitle"
                        variant="subtitle1"
                        data-testid={`homecardPSub_${item}`}
                      >
                        {item === 'PL'
                          ? 'Perjanjian Lama'
                          : item === 'PB'
                          ? 'Perjanjian Baru'
                          : item === 'ALT'
                          ? 'Tambahan'
                          : ''}
                      </Typography>
                    </Grid>
                  </Grid>
                ))}

                <br />
                <Fab
                  className="homecard__button"
                  size="small"
                  variant="extended"
                  color="primary"
                  onClick={toBible}
                  disabled={!data}
                  data-testid="homecardButton"
                >
                  Baca
                </Fab>
              </CardContent>
            </Card>
          </div>
        </div>
      </Fade>
    </>
  )
}

export default Home
