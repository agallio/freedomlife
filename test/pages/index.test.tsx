import { render } from '../testUtils'
import { Home } from '../../pages/index'
import { dayjs } from '../../src/utils'

describe('Home Page', () => {
  describe('Header', () => {
    // it('title rendered as "FreedomLife"', () => {
    //   const { getByTestId } = render(<Home />, {})
    //   const headerTitle = getByTestId('headerTitle')
    //   expect(headerTitle.textContent).toBe('FreedomLife')
    // })

    // Christmas Edition
    it('title rendered as "FreedomLife ðŸŽ„"', () => {
      const { getByTestId } = render(<Home />, {})
      const headerTitle = getByTestId('headerTitle')
      expect(headerTitle.textContent).toBe('FreedomLife ðŸŽ„')
    })

    it('subtitle rendered as "Aplikasi Panduan Baca Alkitab Setahun"', () => {
      const { getByTestId } = render(<Home />, {})
      const headerSubtitle = getByTestId('headerSubtitle')
      expect(headerSubtitle.textContent).toBe(
        'Aplikasi Panduan Baca Alkitab Setahun'
      )
    })
  })

  describe('Card', () => {
    it('title rendered as "Panduan Hari Ini"', () => {
      const { getByTestId } = render(<Home />, {})
      const homecardTitle = getByTestId('homecardTitle')
      expect(homecardTitle.textContent).toBe('Panduan Hari Ini')
    })

    it(`subtitle rendered as "${dayjs().format('dddd, DD MMMM YYYY')}"`, () => {
      const { getByTestId } = render(<Home />, {})
      const homecardSubtitle = getByTestId('homecardSubtitle')
      expect(homecardSubtitle.textContent).toBe(
        dayjs().format('dddd, DD MMMM YYYY')
      )
    })

    it('pbox 1st title is "PL"', () => {
      const { getByTestId } = render(<Home />, {})
      const homecardPBox_PL = getByTestId('homecardPBox_PL')
      expect(homecardPBox_PL.textContent).toBe('PL')
    })

    it('pbox 2nd title is "PB"', () => {
      const { getByTestId } = render(<Home />, {})
      const homecardPBox_PB = getByTestId('homecardPBox_PB')
      expect(homecardPBox_PB.textContent).toBe('PB')
    })

    it('pbox 3rd title is "ALT"', () => {
      const { getByTestId } = render(<Home />, {})
      const homecardPBox_ALT = getByTestId('homecardPBox_ALT')
      expect(homecardPBox_ALT.textContent).toBe('ALT')
    })

    it('psub 1st value is "Perjanjian Lama"', () => {
      const { getByTestId } = render(<Home />, {})
      const homecardPSub_PL = getByTestId('homecardPSub_PL')
      expect(homecardPSub_PL.textContent).toBe('Perjanjian Lama')
    })

    it('psub 2nd value is "Perjanjian Baru"', () => {
      const { getByTestId } = render(<Home />, {})
      const homecardPSub_PB = getByTestId('homecardPSub_PB')
      expect(homecardPSub_PB.textContent).toBe('Perjanjian Baru')
    })

    it('psub 3rd value is "Tambahan"', () => {
      const { getByTestId } = render(<Home />, {})
      const homecardPSub_ALT = getByTestId('homecardPSub_ALT')
      expect(homecardPSub_ALT.textContent).toBe('Tambahan')
    })

    it('button value is "Baca"', () => {
      const { getByTestId } = render(<Home />, {})
      const homecardButton = getByTestId('homecardButton')
      expect(homecardButton.textContent).toBe('Baca')
    })
  })
})
