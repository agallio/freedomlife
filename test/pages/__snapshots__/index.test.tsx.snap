// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page matches snapshot 1`] = `
<DocumentFragment>
  <div
    __source="[object Object]"
    class="MuiLinearProgress-root MuiLinearProgress-colorSecondary MuiLinearProgress-indeterminate"
    role="progressbar"
  >
    <div
      class="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Indeterminate"
    />
    <div
      class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorSecondary"
    />
  </div>
  <div
    __source="[object Object]"
    class="container"
    style="opacity: 1; padding-bottom: 90px; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <h2
      __source="[object Object]"
      class="header__title"
      data-testid="headerTitle"
      style="margin-top: 65px;"
    >
      FreedomLife
    </h2>
    <p
      __source="[object Object]"
      class="header__subtitle"
      data-testid="headerSubtitle"
    >
      Aplikasi Panduan Baca Alkitab Setahun
    </p>
    <div
      __source="[object Object]"
      style="margin-top: 40px;"
    >
      <div
        __source="[object Object]"
        class="MuiPaper-root MuiCard-root homecard MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          __source="[object Object]"
          class="MuiCardContent-root"
        >
          <p
            __source="[object Object]"
            class="MuiTypography-root homecard__title MuiTypography-body1"
            data-testid="homecardTitle"
          >
            Panduan Hari Ini
          </p>
          <p
            __source="[object Object]"
            class="MuiTypography-root homecard__subtitle MuiTypography-body1"
            data-testid="homecardSubtitle"
          >
            Jumat, 20 November 2020
          </p>
          <div
            __source="[object Object]"
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center"
            style="margin-top: 5px;"
          >
            <div
              __source="[object Object]"
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-2 MuiGrid-grid-md-2"
            >
              <div
                __source="[object Object]"
                class="homecard__pbox"
              >
                <p
                  __source="[object Object]"
                  class="homecard__pbox__text"
                  data-testid="homecardPBox_PL"
                >
                  PL
                </p>
              </div>
            </div>
            <div
              __source="[object Object]"
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 MuiGrid-grid-sm-10 MuiGrid-grid-md-10"
            >
              <span
                __source="[object Object]"
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
                style="height: 25px;"
              />
              <p
                __source="[object Object]"
                class="MuiTypography-root homecard__subtitle MuiTypography-body1"
                data-testid="homecardPSub_PL"
              >
                Perjanjian Lama
              </p>
            </div>
          </div>
          <div
            __source="[object Object]"
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center"
            style="margin-top: 5px;"
          >
            <div
              __source="[object Object]"
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-2 MuiGrid-grid-md-2"
            >
              <div
                __source="[object Object]"
                class="homecard__pbox"
              >
                <p
                  __source="[object Object]"
                  class="homecard__pbox__text"
                  data-testid="homecardPBox_PB"
                >
                  PB
                </p>
              </div>
            </div>
            <div
              __source="[object Object]"
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 MuiGrid-grid-sm-10 MuiGrid-grid-md-10"
            >
              <span
                __source="[object Object]"
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
                style="height: 25px;"
              />
              <p
                __source="[object Object]"
                class="MuiTypography-root homecard__subtitle MuiTypography-body1"
                data-testid="homecardPSub_PB"
              >
                Perjanjian Baru
              </p>
            </div>
          </div>
          <div
            __source="[object Object]"
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center"
            style="margin-top: 5px;"
          >
            <div
              __source="[object Object]"
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-2 MuiGrid-grid-md-2"
            >
              <div
                __source="[object Object]"
                class="homecard__pbox"
              >
                <p
                  __source="[object Object]"
                  class="homecard__pbox__text"
                  data-testid="homecardPBox_ALT"
                >
                  ALT
                </p>
              </div>
            </div>
            <div
              __source="[object Object]"
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 MuiGrid-grid-sm-10 MuiGrid-grid-md-10"
            >
              <span
                __source="[object Object]"
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
                style="height: 25px;"
              />
              <p
                __source="[object Object]"
                class="MuiTypography-root homecard__subtitle MuiTypography-body1"
                data-testid="homecardPSub_ALT"
              >
                Tambahan
              </p>
            </div>
          </div>
          <br
            __source="[object Object]"
          />
          <button
            __source="[object Object]"
            class="MuiButtonBase-root MuiFab-root homecard__button MuiFab-extended MuiFab-sizeSmall Mui-disabled MuiFab-primary Mui-disabled"
            data-testid="homecardButton"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiFab-label"
            >
              Baca
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
